larpandoracontent/LArCheating/CheatingPfoCreationAlgorithm.cc

```diff
@@ -129,6 +130,27 @@ void CheatingPfoCreationAlgorithm::CreatePfos(const MCParticleToClusterListMap &
 
             const ParticleFlowObject *pPfo(nullptr);
             PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::ParticleFlowObject::Create(*this, pfoParameters, pPfo));
+
+            if (m_addVertices)
+            {
+                const VertexList *pVertexList = NULL; std::string vertexListName;
+                PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::CreateTemporaryListAndSetCurrent(*this, pVertexList, vertexListName));
+
+                PandoraContentApi::Vertex::Parameters parameters;
+                parameters.m_position = CartesianVector(pMCParticle->GetVertex().GetX(), pMCParticle->GetVertex().GetY(), pMCParticle->GetVertex().GetZ());
+                parameters.m_vertexLabel = VERTEX_START;
+                parameters.m_vertexType = VERTEX_3D;
+
+                const Vertex *pVertex(NULL);
+                PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::Vertex::Create(*this, parameters, pVertex));
+
+                if (!pVertexList->empty())
+                {
+                    PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::SaveList<Vertex>(*this, m_vertexListName));
+                    PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::ReplaceCurrentList<Vertex>(*this, m_vertexListName));
```

Comment:
Might want to add option as to whether you really want to replace the current list. Basically, and as for most of the above, suggest that you just follow what is already done for the Pfos.